server/expression/in_tuple.go

```diff
@@ -94,16 +99,33 @@ func (it *InTuple) Eval(ctx *sql.Context, row sql.Row) (any, error) {
 	for i, rightValue := range rightValues {
 		it.arrayLiterals[i].value = rightValue
 	}
+
 	// Now we can loop over all of the comparison functions, as they'll reference their respective values
+	// The rules for null comparisons are subtle: an IN expression that includes a NULL in the tuple will return null
+	// instead of false if a match is not found, but true otherwise.
+	sawNull := false
 	for _, compFunc := range it.compFuncs {
 		result, err := compFunc.Eval(ctx, row)
 		if err != nil {
 			return nil, err
 		}
-		if result.(bool) {
-			return true, nil
+
+		if result == nil {
+			sawNull = true
+		} else {
+			isEqual, isBool := result.(bool)
+			if isEqual {
+				return true, nil
+			} else if !isBool {
+				return false, fmt.Errorf("%T: expected comparison function to return a bool but returned `%T`", it, result)
+			}
```

Comment:
This ordering seems weird. I get that `isEqual` will only be `true` when `isBool` is `true`, but it's kind of a...standard to check the cast first? But this way is technically more performant since we're excluding a check. It's worth leaving, but seems weird lol.

Reply:
I'll comment